kotlin/export-tiles/src/main/java/com/example/exporttiles/MainActivity.kt

```diff
@@ -0,0 +1,229 @@
+      setViewpoint(Viewpoint(51.5, 0.0, 10000000.0))
+      // add the graphics overlay to the map view
+      graphicsOverlays.add(graphicsOverlay)
+    }
+
+    // update the box whenever the viewpoint changes
+    mapView.addViewpointChangedListener {
+      if (mapView.map.loadStatus == LoadStatus.LOADED) {
+        // upper left corner of the downloaded tile cache area
+        val minScreenPoint: android.graphics.Point = android.graphics.Point(150, 175)
+        // lower right corner of the downloaded tile cache area
+        val maxScreenPoint: android.graphics.Point = android.graphics.Point(
+          mapView.getWidth() - 150,
+          mapView.getHeight() - 250
+        )
+        // convert screen points to map points
+        val minPoint: Point? = mapView?.screenToLocation(minScreenPoint)
+        val maxPoint: Point? = mapView?.screenToLocation(maxScreenPoint)
+        // use the points to define and return an envelope
+        downloadArea.geometry = Envelope(minPoint, maxPoint)
+      }
+    }
+
+    // create up a temporary directory in the app's cache for saving exported tiles
+    val exportTilesDirectory = File(cacheDir, getString(R.string.tile_cache_folder))
+
+    // when the button is clicked, export the tiles to a temporary file
+    exportTilesButton.setOnClickListener {
+      val exportTileCacheTask = ExportTileCacheTask(tiledLayer.uri)
+      val parametersFuture: ListenableFuture<ExportTileCacheParameters> =
+        exportTileCacheTask.createDefaultExportTileCacheParametersAsync(
+          downloadArea.geometry,
+          mapView.mapScale,
+          tiledLayer.maxScale
+        )
+
+      parametersFuture.addDoneListener {
+        val parameters: ExportTileCacheParameters = parametersFuture.get()
+        // export tiles to temporary cache on device
+        val exportTileCacheJob =
+          exportTileCacheTask.exportTileCache(parameters, exportTilesDirectory.path + "file.tpk")
+        exportTileCacheJob.start()
+        // create and show a progress dialog to show export tile progress
+        val dialog = createProgressDialog(exportTileCacheJob)
+        dialog.show()
+
+        // show progress of the export tile cache job on the progress bar
+        exportTileCacheJob.addProgressChangedListener {
+          dialog.progressBar.progress = exportTileCacheJob.progress
+        }
+
+        exportTileCacheJob.addJobDoneListener {
+          dialog.dismiss()
+          if (exportTileCacheJob.status == Job.Status.SUCCEEDED) {
+            val exportedTileCacheResult: TileCache = exportTileCacheJob.result
+            showMapPreview(exportedTileCacheResult)
+          } else {
+            Log.e(TAG, exportTileCacheJob.error.additionalMessage)
+            Toast.makeText(
+              this, exportTileCacheJob.error.additionalMessage, Toast.LENGTH_LONG
+            ).show()
+          }
+        }
+      }
+    }
+
+    // close the preview window
+    closeButton.setOnClickListener { clearPreview() }
```

Comment:
we can use databinding for this:

add `android:onClick="clearPreview"` to the button in xml

and change the private `fun clearPreview()` method to `clearPreview(view: View)`